xgboost_ray/main.py

```diff
@@ -342,26 +361,24 @@ def _train(params: Dict,
     # Train
     fut = [
         actor.train.remote(rabit_args, params, dtrain, evals, *args, **kwargs)
-        for (actor, _) in actors
+        for actor in actors
     ]
 
-    callback_returns = [list() for _ in range(len(actors))]
+    callback_returns = []
     try:
         ready, not_ready = ray.wait(fut, num_returns=len(fut), timeout=0)
         while not_ready:
-            for i, (actor, queue) in enumerate(actors):
-                while not queue.empty():
-                    item = queue.get()
-                    if isinstance(item, Callable):
-                        item()
-                    else:
-                        callback_returns[i].append(item)
+            while not queue.empty():
+                item = queue.get()
+                if isinstance(item, Callable):
+                    item()
+                else:
+                    callback_returns.append(item)
```

Comment:
Wait, this will move callback returns into one global list, right? I think we might actually want to keep them separated by actor rank. There's an update for this code in the current master

Reply:
Is it the user's responsibility to to make sure that their callback returns a tuple with the first element being the actor rank?

Reply:
Fixed.